src/JsFunctionsScanner.php

```diff
@@ -0,0 +1,176 @@
+	}
+
+	/**
+	 * Disable comments extraction.
+	 */
+	public function disableCommentsExtraction() {
+		$this->extractComments = false;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function saveGettextFunctions( Translations $translations, array $options ) {
+		$ast = Peast::latest( $this->code, [
+			'sourceType' => Peast::SOURCE_TYPE_MODULE,
+			'comments'   => false !== $this->extractComments,
+			'jsx'        => true,
+		] )->parse();
+
+		$traverser = new Traverser();
+
+		$all_comments = [];
+
+		/**
+		 * Traverse through JS code to find and extract gettext functions.
+		 *
+		 * Make sure translator comments in front of variable declarations
+		 * and inside nested call expressions are available when parsing the function call.
+		 */
+		$traverser->addFunction( function ( $node ) use ( &$translations, $options, &$all_comments ) {
+			$functions = $options['functions'];
+			$file      = $options['file'];
+
+			/* @var Node $node */
+			foreach( $node->getLeadingComments() as $comment ) {
+				$all_comments[] = $comment;
+			}
+
+			/* @var CallExpression $node */
+			if ( 'CallExpression' !== $node->getType() || 'Identifier' !== $node->getCallee()->getType() ) {
+				return;
+			}
+
+			/* @var CallExpression $node */
+			foreach ( $node->getArguments() as $argument ) {
+				// Support nested function calls.
+				$argument->setLeadingComments( $argument->getLeadingComments() + $node->getLeadingComments() );
+			}
+
+			/* @var Identifier $callee */
+			$callee = $node->getCallee();
+
+			if ( ! isset( $functions[ $callee->getName() ] ) ) {
+				return;
+			}
+
+			foreach( $callee->getLeadingComments() as $comment ) {
+				$all_comments[] = $comment;
+			}
+
+			$domain = $context = $original = $plural = null;
+			$args   = [];
+
+			/* @var Node $argument */
+			foreach ( $node->getArguments() as $argument ) {
+				foreach( $argument->getLeadingComments() as $comment ) {
+					$all_comments[] = $comment;
+				}
+
+				if ( 'Identifier' === $argument->getType() ) {
+					$args[] = ''; // The value doesn't matter as it's unused.
+				}
+
+				if ( 'Literal' === $argument->getType() ) {
+					/* @var Literal $argument */
+					$args[] = $argument->getValue();
+				}
+			}
+
+			switch ( $functions[ $callee->getName() ] ) {
+				case 'text_domain':
+				case 'gettext':
+					if ( ! isset( $args[1] ) ) {
+						break;
+					}
+
+					list( $original, $domain ) = $args;
+					break;
+
+				case 'text_context_domain':
+					if ( ! isset( $args[2] ) ) {
+						break;
+					}
+
+					list( $original, $context, $domain ) = $args;
+					break;
+
+				case 'single_plural_number_domain':
+					if ( ! isset( $args[3] ) ) {
+						break;
+					}
+
+					list( $original, $plural, $number, $domain ) = $args;
+					break;
+
+				case 'single_plural_number_context_domain':
+					if ( ! isset( $args[4] ) ) {
+						break;
+					}
+
+					list( $original, $plural, $number, $context, $domain ) = $args;
+					break;
+			}
+
+			// Todo: Require a domain?
```

Comment:
Should we discuss this? Is there a technical difficulty? What is blocking this?

Reply:
This is some copy/paste leftover from the original implementation, not something new. It's better read as `Todo: Should we really require a domain here?`.

Some background can be found in #7.

Basically, right now a text domain is required. It's not possible to not use one.

I now opened #35 to start a new discussion around this and will remove the comment in the mean time. It doesn't really communicate what the issue at hand is.